src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -26,17 +30,101 @@ export class AwsSamDebugConfigurationProvider implements vscode.DebugConfigurati
         token?: vscode.CancellationToken
     ): Promise<AwsSamDebuggerConfiguration | undefined> {
         if (!AWS_SAM_DEBUG_REQUEST_TYPES.has(debugConfiguration.request)) {
-            vscode.window.showInformationMessage('Invalid request type')
```

Comment:
Instead of sprinkling the showErrorMessagr everywhere, why not return the error message and if not undefined, call it?